pkg/v22/resource/app/desired.go

```diff
@@ -36,6 +41,54 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
 		return nil, microerror.Mask(err)
 	}
 
+	// TODO: Remove connection to tenant cluster once all tenant clusters use
+	// app CRs instead of chartconfig CRs.
+	//
+	//	https://github.com/giantswarm/giantswarm/issues/7402
+	//
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if cc.Client.TenantCluster.G8s == nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant clients not available")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
```

Comment:
is it really enough with 3 seconds?

Reply:
Resources should execute quickly because some control planes have a lot of tenant clusters.

This is for a single request to the tenant API server. If it takes longer we cancel and retry.